src/scripts/import-sdr-shipments-sheet/add-countries.ts

```diff
@@ -0,0 +1,260 @@
+import qs from "qs";
+import lookup from "country-code-lookup";
+import { STRAPI_ENV } from "../strapi-env";
+import {
+  Country,
+  ShipmentCsv,
+  CountryUploadWorkflow,
+  UploadWorkflowStatus,
+  CountryUploadWorkflowResults,
+} from "./types.d";
+
+/* Add Countries From SDR Shipments Sheet
+ * ========================================================================== */
+export async function addCountries(shipments: ShipmentCsv[]) {
+  console.log("Adding Geo.Countries from SDR Shipments sheet...");
+
+  const uniqueCountries = [...consolidateCountries(shipments)];
+
+  const results = await Promise.allSettled<CountryUploadWorkflow>(
+    uniqueCountries.map((country) => {
+      return new Promise<CountryUploadWorkflow>((resolve, _reject) => {
+        resolve({
+          data: {},
+          orig: country,
+          status: UploadWorkflowStatus.PROCESSING,
+          logs: [],
+        });
+      })
+        .then(parseCountry)
+        .then(getCountry)
+        .then(uploadCountry);
+    }),
+  );
+
+  // { "SUCCESS": [], "ALREADY_EXISTS": [], ...}
+  const resultsMap: CountryUploadWorkflowResults = Object.keys(
+    UploadWorkflowStatus,
+  ).reduce((resultsMap, key) => {
+    resultsMap[key] = [];
+    return resultsMap;
+  }, {} as CountryUploadWorkflowResults);
+
+  results
+    .map((result) => {
+      if (isFulfilled(result)) {
+        return result.value;
+      } else {
+        return result.reason;
+      }
+    })
+    .reduce((resultsMap, workflowResult) => {
+      if (workflowResult.status) {
+        resultsMap[workflowResult.status].push(workflowResult);
+      } else {
+        resultsMap[UploadWorkflowStatus.OTHER].push(workflowResult);
+      }
+      return resultsMap;
+    }, resultsMap);
```

Comment:
With lodash this could be:

```typescript
const resultsMap = {
  ...chain(UploadWorkflowStatus)
         .mapKeys(key => [key, []])
         .fromPairs()
         .value(),
  ...chain(results)
         .map(res => isFulfilled(res) ? res.value : res.reason)
         .groupBy(res => res.status ?? UploadWorkflowStatus.OTHER)
         .value()
}
```